### PR TITLE
LPS-170313 auto SF

### PR DESCRIPTION
**Pull request by seltong**
Related issue: https://issues.liferay.com/browse/LPS-170313

### CODE REVIEW

**Comment by seltong** [comment]
This pull request is blocked by: [LPS-175202](https://issues.liferay.com/browse/LPS-175202) and [LPS-180237](https://issues.liferay.com/browse/LPS-180237)

**Comment by samuelromeiro** [comment]
Hey @seltong, can you check the conflicts of this PR?

**Comment by seltong** [comment]
@samuelromeiro sure, I'm working on it :+1:

**Comment by seltong** [comment]
ci:test:sf

**Comment by samuelromeiro** [comment]
@seltong, both tickets ([LPS-175202](https://issues.liferay.com/browse/LPS-175202) and [LPS-180237](https://issues.liferay.com/browse/LPS-180237)) are close. Can you review the PR and send to Brian?

**Comment by seltong** [comment]
@samuelromeiro I'm working on it

**Comment by samuelromeiro** [comment]
Hey @seltong, Pedro's PR was merged. 
Can you check de conflicts and resend, please?

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:object

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/action/engine/ObjectActionEngineImpl.java [hunk 1]
```diff
@@ -181,18 +184,54 @@ private void _executeObjectAction(
...
+			if ((StringUtil.equals(
+					objectAction.getObjectActionExecutorKey(),
+					ObjectActionExecutorConstants.KEY_UPDATE_OBJECT_ENTRY) &&
+				 SetUtil.isNotEmpty(objectEntryIds) &&
+				 objectEntryIds.contains(objectEntryId)) ||
+				(!StringUtil.equals(
+					objectAction.getObjectActionExecutorKey(),
+					ObjectActionExecutorConstants.KEY_UPDATE_OBJECT_ENTRY) &&
+				 objectActionIds.containsKey(
+					 objectAction.getObjectActionId())) ||
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/action/engine/ObjectActionEngineImpl.java:226]: @seltong can we simplify this code to somehting like

```
Map<Long, Set<Long>> objectActionIds =
	ObjectActionThreadLocal.getObjectActionIds();

if (!StringUtil.equals(
		objectAction.getObjectActionExecutorKey(),
		ObjectActionExecutorConstants.KEY_UPDATE_OBJECT_ENTRY) &&
	objectActionIds.containsKey(objectAction.getObjectActionId())) {

	// This return avoids unnecessary logic as we don't need to 
    // calculate the objectEntryId in this case. 
	
    return;
}

long objectEntryId = _getObjectEntryId(
	objectDefinition, payloadJSONObject);

if (StringUtil.equals(
		objectAction.getObjectActionExecutorKey(),
		ObjectActionExecutorConstants.KEY_UPDATE_OBJECT_ENTRY)) {

	Set<Long> objectEntryIds = objectActionIds.get(
		objectAction.getObjectActionId());

	if (SetUtil.isNotEmpty(objectEntryIds) &&
		objectEntryIds.contains(objectEntryId)) {

		return;
	}
}

try {
	if (!_evaluateConditionExpression(
			objectAction.getConditionExpression(), variables)) {

		return;
	}
....
```

```
private long _getObjectEntryId(
	ObjectDefinition objectDefinition, JSONObject payloadJSONObject) {

	if (objectDefinition.isUnmodifiableSystemObject()) {
		Map<String, Object> map =
			(Map<String, Object>)payloadJSONObject.get(
				"model" + objectDefinition.getName());

		return (long)map.get(objectDefinition.getPKObjectFieldName());
	}

	Map<String, Object> map = (Map<String, Object>)payloadJSONObject.get(
		"objectEntry");

	return GetterUtil.getLong(
		map.get("id"), (long)map.get("objectEntryId"));
}
```
?

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
@seltong please see https://github.com/liferay-objects/liferay-portal/pull/2020

**Pull request closed without merging**